src/main/java/com/faforever/client/game/GamesController.java

```diff
@@ -147,18 +164,35 @@ void postConstruct() {
     filteredItems = new FilteredList<>(gameService.getGameInfoBeans());
     filteredItems.setPredicate(OPEN_GAMES_PREDICATE);
 
-    // FIXME remove after switching between tiles/tables is implemented
     onDetailsButtonPressed();
-    switchViewButton.setVisible(false);
-    switchViewButton.setManaged(false);
   }
 
   public void displayGameDetail(GameInfoBean gameInfoBean) {
-    mapImageView.setImage(mapService.loadLargePreview(gameInfoBean.getMapName()));
+    currentGameInfoBean = gameInfoBean;
     gameTitleLabel.setText(gameInfoBean.getTitle());
+    mapImageView.setImage(mapService.loadLargePreview(gameInfoBean.getMapName()));
+
+    gameInfoBean.mapNameProperty().addListener((observable, oldValue, newValue) -> {
+      gameTitleLabel.setText(newValue);
+      mapImageView.setImage(mapService.loadLargePreview(newValue));
+    });
+
     numberOfPlayersLabel.setText(i18n.get("game.detail.players.format", gameInfoBean.getNumPlayers(), gameInfoBean.getMaxPlayers()));
-    hosterLabel.setText(gameInfoBean.getHost());
-    gameModeLabel.setText(gameInfoBean.getFeaturedMod());
+    hosterLabel.textProperty().bind(gameInfoBean.hostProperty());
+    gameModeLabel.textProperty().bind(gameInfoBean.featuredModProperty());
+    mapLabel.textProperty().bind(gameInfoBean.mapNameProperty());
+    createTeams(gameInfoBean.getTeams());
+  }
+
+  private void createTeams(ObservableMap<? extends String, ? extends List<String>> teamsList) {
```

Comment:
`teamsList` suggests that it is a list of teams. But since it's a map I'd call it `playersByTeamNumber`